### PR TITLE
Add hexgrad/Kokoro-82M loader (StyleTTS2 TTS / iSTFTNet vocoder)

### PR DESCRIPTION
**Pull request by saiarthiraguram**
### Ticket 
[Github Issue](https://github.com/tenstorrent/tt-xla/issues/5215)

### Problem description
`hexgrad/Kokoro-82M` (StyleTTS2-derived TTS: PL-BERT prosody encoder + duration
predictor + iSTFTNet vocoder, ~82M params) had no tt-forge-models loader, so it
couldn't be brought up on Tenstorrent hardware. This PR adds the model + a
tensors-only `ForgeModel` loader for compile and PCC bringup.

### What's changed
Vendored the Kokoro model under `kokoro/pytorch/src/` (adapted from
[StyleTTS2](https://github.com/yl4579/StyleTTS2), Apache-2.0; upstream
attribution retained) and added a `ForgeModel` loader.

- **`pytorch/loader.py`** — builds `KModel` from `config.json`, fp32-pinned (the
  iSTFTNet vocoder is numerically sensitive), `disable_complex=True` (conv-based
  `CustomSTFT`, no complex ops → compiler-friendly). `load_weights=True` loads
  the trained `kokoro-v1_0.pth` (required for meaningful PCC).
  `DEFAULT_NUM_TOKENS=8` keeps the inverse-STFT graph tractable.
  `_KokoroWrapper` exposes a clean `forward(input_ids, ref_s) -> waveform` (the
  stock `KModel.forward` takes a phoneme string).
- **`src/model.py`** — `load_weights` flag; round-half-up via `floor(x + 0.5)`
  (avoids `stablehlo.round_nearest_even`, which has no tt-mlir legalization);
  fixed-duration alignment (`_FIXED_DUR`) so the output waveform length is
  deterministic on device vs. CPU (the bf16 duration logits otherwise round
  across `floor()` boundaries differently and produce mismatched lengths).
- **`src/modules.py`** — `_run_lstm` helper that bypasses `pack_padded_sequence`
  (the `PackedSequence` is not materialized by the XLA tracer); dtype-matched
  zero padding on the LSTM path.
- **`src/istftnet.py`** — deterministic source module (zeroed RNG dither so the
  CPU golden and TT passes are comparable) and dtype casts on the
  sine-source → linear path.
- **`src/custom_stft.py`** — vendored conv-based STFT/iSTFT (verbatim).

**Bringup status:** compiles and runs end-to-end on n150 with trained weights,
producing a finite, sane-magnitude waveform. It does **not** yet reach
PCC 0.99 — the iSTFTNet vocoder (sine-phase `cumsum` + STFT/iSTFT over a long
1-D waveform) is sensitive to bf16 tensor **storage**, capping waveform PCC at
~0.14 (decoder output is already 0.948; `fp32_dest_acc_en`+hifi4 is
bit-identical, i.e. storage-bound not accumulation-bound). Tracked and marked
`KNOWN_FAILURE_XFAIL` in tt-xla via tenstorrent/tt-xla#5332.

### Checklist
- [x] New/Existing tests provide coverage for changes — discoverable as
  `kokoro/pytorch-hexgrad/Kokoro-82M-single_device-inference`; runs E2E on n150.

## Logs
[model.log](https://github.com/user-attachments/files/29249201/iter_n150_13_fp32acc_realw_run.log)

### CODE REVIEW

**Review by nvukobratTT — COMMENTED** [review]
A general comment: why don't we just submodule a third-party GitHub project and load the respective components as before?

File: kokoro/pytorch/src/modules.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+                    device=x.device,
+                    dtype=x.dtype,
+                )
+                x_pad[:, :, : x.shape[-1]] = x
+                x = x_pad
+
+        return x.transpose(-1, -2)
+
+
+# https://github.com/yl4579/StyleTTS2/blob/main/Utils/PLBERT/util.py
```
Comment (nvukobratTT) [inline kokoro/pytorch/src/modules.py]: Is it possible just to include this model as GitHub third_party instead of copy-pasting the whole model?

Comment (saiarthiraguram) [reply kokoro/pytorch/src/modules.py]: Sure Nikola, will make the required changes

Comment (saiarthiraguram) [reply kokoro/pytorch/src/modules.py]: Added in the recent commit 

## Logs
[test_runxfail_fix2_20260701_155139.log](https://github.com/user-attachments/files/29560046/test_runxfail_fix2_20260701_155139.log)

**Comment by saiarthiraguram** [comment]
> A general comment: why don't we just submodule a third-party GitHub project and load the respective components as before?

Refactored the code to use PYPI and added patches to be applied on top in the recent commit. Please lmk if this is fine
